File: Top 100 Codes/02-Working with Numbers/17-Can a number be expressed as a sum of two prime numbers/Cananumberbeexpressedasasumoftwoprimenumbers.cpp
#include <iostream>
using namespace std;

// Function to check if a number is prime
bool isPrime(int num) {
    if (num <= 1) 
        return false;
    for (int i = 2; i * i <= num; i++) {
        if (num % i == 0) return false;
    }
    return true;
}

// Function to check if a number can be expressed as sum of two prime numbers
void checkSumOfTwoPrimes(int n) {
    bool found = false;
    for (int i = 2; i <= n / 2; i++) 
    {
        if (isPrime(i) && isPrime(n - i)) 
        {
            cout << n << " = " << i << " + " << (n - i) << endl;
            found = true;
        }
    }
    if (!found)
        cout << n << " cannot be expressed as the sum of two prime numbers." << endl;
}

int main() {
    int num;
    cout << "Enter a number: ";
    cin >> num;
    checkSumOfTwoPrimes(num);
    return 0;
}

File: Top 100 Codes/02-Working with Numbers/17-Can a number be expressed as a sum of two prime numbers/Cananumberbeexpressedasasumoftwoprimenumbers.py
def is_prime(num):
    if num <= 1:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True

def check_sum_of_two_primes(n):
    found = False
    for i in range(2, n // 2 + 1):
        if is_prime(i) and is_prime(n - i):
            print(f"{n} can be expressed as the sum of {i} and {n - i}")
            found = True
    if not found:
        print(f"{n} cannot be expressed as the sum of two prime numbers.")

# Taking user input
num = int(input("Insert the num: "))
check_sum_of_two_primes(num)
